config-examples/fail2ban-jail.cf

```diff
@@ -1,21 +1,37 @@
 [DEFAULT]
 
 # "bantime" is the number of seconds that a host is banned.
-#bantime  = 10m
+bantime = 3h
 
 # A host is banned if it has generated "maxretry" during the last "findtime"
 # seconds.
-#findtime  = 10m
+findtime = 10m
 
 # "maxretry" is the number of failures before a host get banned.
-#maxretry = 5
+maxretry = 3
 
 # "ignoreip" can be a list of IP addresses, CIDR masks or DNS hosts. Fail2ban
 # will not ban a host which matches an address in this list. Several addresses
 # can be defined using space (and/or comma) separator.
-#ignoreip = 127.0.0.1/8
+ignoreip = 127.0.0.1/8
 
-# Default ban action
-# nftables-multiport:	block IP only on affected port
-# nftables-allports:	block IP on all ports
-#banaction = nftables-allports
+# default ban action
+# nftables-multiport: block IP only on affected port
+# nftables-allports:  block IP on all ports
+banaction = nftables-allports
+
+[dovecot]
+enabled = true
+
+[postfix]
+enabled = true
+
+[postfix-sasl]
+enabled = true
+
+# This jail is used for manual bans.
+# To ban an IP address use: setup.sh fail2ban ban <IP>
+[custom]
+enabled = true
+bantime = 180d
+port = smtp,pop3,pop3s,imap,imaps,submission,465,sieve
```

Comment:
```suggestion
port = smtp,pop3,pop3s,imap,imaps,submission,submissions,sieve
```

Might as well use the service mapping like the others?:

```console
$ grep submissions /etc/services
submissions     465/tcp         ssmtp smtps urd # Submission over TLS [RFC8314]
```

Reply:
This was on purpose. Let me explain why I decided to do it this way:

I have aggregated the port line from the ports used in the jails `postfix`, `postfix-sasl` and `dovecot`. For some reason, in fail2bans default config, all mentioned jails use the numeric value `465` instead of `submissions`:

```
[postfix]
# To use another modes set filter parameter "mode" in jail.local:
mode    = more
port    = smtp,465,submission
logpath = %(postfix_log)s
backend = %(postfix_backend)s
```

I thought this was maybe done for some reason (to prevent an issue?).
Instead of streamlining this, I went with sticking to the default config.

Reply:
> I have aggregated the port line from the ports used in the jails `postfix`, `postfix-sasl` and `dovecot`. For some reason, in fail2bans default config, all mentioned jails use the numeric value `465` instead of `submissions`

They may just be playing it safe if the config is distro-agnostic.

In a PR I was making in 2020 regarding fail2ban at the time, we had outdated configs for `imap2` or something iirc, which I think was related to earlier Ubuntu base image at the time maybe missing `imap`.. I think I detailed it in the PR.

AFAIK, these all reference `/etc/services` to lookup their port by name or alias.

You could just provide all ports instead of their service names too, personally I think we should try make it consistent. I don't see it being a problem unless we switch base images in future and for some reason the service isn't defined, we don't have a test for such, but we already use the outdated `smtps` service (now and alias) in Postfix `master.cf` (_and two doc examples with an `haproxy` line_):

https://github.com/docker-mailserver/docker-mailserver/blob/3cb5668b6f62258d8c21629c0f5e649085e4f0b6/target/postfix/master.cf#L30

As per [Postfix `master.cf` docs for the `inet` service type](http://www.postfix.org/master.5.html):

> The service name is specified as host:port, denoting the host and port on which new connections should be accepted. The host part (and colon) may be omitted.
> Either host or port may be given in symbolic form (see hosts(5) or services(5)) or in numeric form (IP address or port number).

---

> I thought this was maybe done for some reason (to prevent an issue?).

`465` was previously `smtps` which would be misleading now, we already know that those not working with mail servers as well as some outdated advice out there already can make that port a bit confusing for some.

Chances are they chose `465` just to be on the safe-side, perhaps some distros did not have `submissions` service in `/etc/services`, especially at the time that was contributed to fail2ban. Probably nobody has bothered or had reason to adjust it since (_I haven't gone looking through fail2ban history to verify_).

I do recall `/etc/services` being inconsistent across some distros when I originally wrote a PR in 2020 for `docker-mailserver` (didn't merge though).